dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LoginPropertyTest.java

```diff
@@ -0,0 +1,329 @@
+     *
+     * Additionally, we expect the user display name to be SN because loginProperty
+     * takes precedence.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testLoginPropertyWithFilter() throws Exception {
+        updateLibertyServer(true, "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertNotNull("Should find CWIML4505W message.", libertyServer.waitForStringInLog("CWIML4505W"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userCn, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userUid, "password"));
+        assertEquals(userSn, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Configuration:
+     *
+     * UserFilter: not explicitly configured
+     * LoginProperties: SN
+     *
+     * Expected Results:
+     *
+     * User can login with SN, but not with CN or UID.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testLoginProperty() throws Exception {
+        updateLibertyServer(false, "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userUid, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userCn, "password"));
+    }
+
+    /**
+     * Configuration:
+     *
+     * UserFilter: not explicitly configured
+     * LoginProperties: CN, SN
+     *
+     * Expected Results:
+     *
+     * User can login with CN and SN, but not with UID.
+     *
+     * Additionally, we expect the user display name to be cn as this is
+     * the first listed loginProperty.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testMultipleLoginProperty() throws Exception {
+        updateLibertyServer(false, "cn", "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userCn, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userUid, "password"));
+        assertEquals(userCn, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Define the loginProperty with a "bad" attribute (not a WIM property)
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testBadLoginProperty() throws Exception {
+        updateLibertyServer(false, "badconfig");
+        assertNotNull("Should find CWIML4505W message.", libertyServer.waitForStringInLog("CWIML4506E"));
```

Comment:
As I mentioned above, maybe have a test where you resolve it by creating the extended property on PersonAccount.